x-pack/legacy/plugins/spaces/server/lib/copy_to_spaces/copy_to_spaces.test.ts

```diff
@@ -0,0 +1,330 @@
+      savedObjectsService
+    );
+
+    const result = await copySavedObjectsToSpaces('sourceSpace', ['destination1', 'destination2'], {
+      includeReferences: true,
+      overwrite: true,
+      objects: [
+        {
+          type: 'dashboard',
+          id: 'my-dashboard',
+        },
+      ],
+    });
+
+    expect(result).toMatchInlineSnapshot(`
+                                                Object {
+                                                  "destination1": Object {
+                                                    "errors": undefined,
+                                                    "success": true,
+                                                    "successCount": 3,
+                                                  },
+                                                  "destination2": Object {
+                                                    "errors": undefined,
+                                                    "success": true,
+                                                    "successCount": 3,
+                                                  },
+                                                }
+                                `);
+
+    expect((savedObjectsService.importExport.getSortedObjectsForExport as jest.Mock).mock.calls)
+      .toMatchInlineSnapshot(`
+                              Array [
+                                Array [
+                                  Object {
+                                    "exportSizeLimit": 1000,
+                                    "includeReferencesDeep": true,
+                                    "namespace": "sourceSpace",
+                                    "objects": Array [
+                                      Object {
+                                        "id": "my-dashboard",
+                                        "type": "dashboard",
+                                      },
+                                    ],
+                                    "savedObjectsClient": Object {
+                                      "errors": Object {
+                                        "isNotFoundError": [Function],
+                                      },
+                                    },
+                                    "types": Array [
+                                      "dashboard",
+                                      "visualization",
+                                    ],
+                                  },
+                                ],
+                              ]
+                    `);
+
+    expect((savedObjectsService.importExport.importSavedObjects as jest.Mock).mock.calls)
+      .toMatchInlineSnapshot(`
+                  Array [
+                    Array [
+                      Object {
+                        "namespace": "destination1",
+                        "objectLimit": 1000,
+                        "overwrite": true,
+                        "readStream": Readable {
```

Comment:
Good points..

`readStreamToCompletion` is tested via the [`getSortedObjectsForExport` snapshot](https://github.com/elastic/kibana/pull/38014/files/3eb982b0ec413f3c8632ace7294ed2d0474f9161#diff-0d234c6d89aa1b90b9a8418c2634b9f9R119), but we could/should add more than one object to the list to ensure the stream is actually read to completion, and not just reading the first entry.


`createReadableStreamFromArray` definitely has a testing gap here....let me give this some more thought on how to best test this

Reply:
good points, we definitely have a testing gap here. Let me give this some more thought.

Reply:
> Are we still concerned about the down-stream import mutating the source? If so, we'll want to verify this here as well.

I don't think so; I haven't been able to reproduce or track down a likely culprit. The API integration tests that verify the "copy to multiple spaces" functionality should provide sufficient coverage, so I don't think anything beyond that is necessary

Reply:
> Using jest's snapshotting here doesn't ensure that we're actually passing a readable stream that contains all of the expected objects. The only coverage that we have of readStreamToCompletion and createReadableStreamFromArray are from its consumers, so we really should be checking this.

I added additional checks to ensure we are reading the stream to completion, and creating a stream from the object array correctly. I kept the snapshot in, because I think it's still beneficial for the other parameters we pass to the import/export service. The stream output is a bit noisy there, so if you'd rather see it written another way, we can do that too.